project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/settings/MinimizationPluginState.kt

```diff
@@ -32,8 +31,16 @@ class MinimizationPluginState : BaseState() {
 
     @Property(surroundWithTag = false)
     @XCollection(style = XCollection.Style.v1, elementName = "minimizationTransformations")
-    var minimizationTransformations: MutableList<TransformationDescriptors> = mutableListOf(
+    var minimizationTransformations: List<TransformationDescriptors> = listOf(
         TransformationDescriptors.PATH_RELATIVIZATION,
     )
-    var exceptionComparingStrategy by enum<ExceptionComparingStrategy>(ExceptionComparingStrategy.SIMPLE)
+
+    // service settings fields
+    var isFrozen: Boolean = false
+    var configMode: StageConfigMode = StageConfigMode.DEFAULT
+    var isFileStageEnabled: Boolean = false
+    var selectedHierarchyStrategy: HierarchyCollectionStrategy = HierarchyCollectionStrategy.FILE_TREE
+    var selectedDDStrategy: DDStrategy = DDStrategy.PROBABILISTIC_DD
+    var isHierarchyStrategyEnabled: Boolean = false
+    var isDDAlgorithmEnabled: Boolean = false
```

Comment:
I guess these settings should be somewhere else, for instance, in `AppSettingsComponent`. Because now they are a little confusing, like: `isFileStageEnables = false`, but at the same time `stages = listOf(FunctionLevelStage(...))`. Looks ambiguous and not synchronized. WDYT?